### PR TITLE
feat(Necord): add presence

### PR DESCRIPTION
**Pull request by SocketSomeone**
## Description 

Add presence for https://necord.org

- Detect Idling
- Reading page with current content and reading process
- Detect selecting categories
- Detect searching and results

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image

image

image

image

image

</details>

### CODE REVIEW

**Comment by SocketSomeone** [comment]
@RisingSunLight42 Hope it looks better now

**Review by RisingSunLight42 — CHANGES_REQUESTED** [review]
For the images' keys, the repo prefer to use an Enumeration containing imgur links for better maintainability.

Also, small suggestions, add a privacy options for the users who don't want to show the things too specific.

But except those points, it's way better than the previous PR 👍

**Comment by SocketSomeone** [comment]
> For the images' keys, the repo prefer to use an Enumeration containing imgur links for better maintainability.

Really? I think change image at discord developers forum most better than use external images, hm...

> Also, small suggestions, add a privacy options for the users who don't want to show the things too specific.

Can you send docs about how to create privacy options?

> But except those points, it's way better than the previous PR 👍

Thank you!

**Comment by SocketSomeone** [comment]
Added privacy settings!

File: websites/N/Necord/metadata.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+			"title": "Show current content",
+			"icon": "fas fa-book",
+			"value": true
+		},
+		{
+			"id": "docs-page-sidebar",
+			"title": "Show selecting category",
+			"icon": "fas fa-book",
+			"value": true
+		},
```
Comment (DarkVillager) [inline websites/N/Necord/metadata.json:46]: Why dont u just combine them into a privacy setting?

File: websites/N/Necord/presence.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+let sidebar: string = null,
+	lastActivity: number = Date.now();
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "https://i.imgur.com/LY804vK.png",
+			startTimestamp: browsingTimestamp,
+		},
+		{ pathname, search } = window.location,
```
Comment (DarkVillager) [inline websites/N/Necord/presence.ts:28]: document is prefered over window. Also add href here.
```suggestion
		{ href, pathname, search } = document.location,
```

File: websites/N/Necord/presence.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+					`(${
+						document.querySelector<HTMLDivElement>("main")?.childElementCount ??
+						0
+					} results)`,
+				].join(" ");
+
+				presenceData.buttons = [
+					{
+						label: "Show Results",
+						url: location.href,
```
Comment (DarkVillager) [inline websites/N/Necord/presence.ts:69]:
```suggestion
						url: href,
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/N/Necord/metadata.json (hunk 1 above)
Comment (SocketSomeone) [reply websites/N/Necord/metadata.json:46]: Added from first example which I saw, so could you send link for documentation about setting's group?

Comment (SocketSomeone) [reply websites/N/Necord/metadata.json:46]: More flexibility?

Comment (RisingSunLight42) [reply websites/N/Necord/metadata.json:46]: So you could add the privacy setting globally and put an if on each to untick them automatically when the privacy is ticked

**Review by RisingSunLight42 — COMMENTED** [review]
I see nothing more to do, just apply the last requested changes !
Now let's wait for the reviewers.

File: websites/N/Necord/metadata.json (hunk 1 above)
Comment (DarkVillager) [reply websites/N/Necord/metadata.json:46]: > So you could add the privacy setting globally and put an if on each to untick them automatically when the privacy is ticked

Agreed.

File: websites/N/Necord/metadata.json [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+	"$schema": "https://schemas.premid.app/metadata/1.7",
+	"author": {
+		"name": "someone",
+		"id": "235413185639874561"
+	},
+	"service": "Necord",
+	"description": {
+		"en": "A module for creating Discord bots using NestJS, based on Discord.js"
+	},
+	"url": "necord.org",
```
Comment (theusaf) [inline websites/N/Necord/metadata.json:11]: Site uses `www` subdomain.
```suggestion
	"url": [
		"necord.org",
		"www.necord.org"
	],
```

File: websites/N/Necord/presence.ts [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+			isSidebarPublic,
+			isIdlingPublic,
+		] = await Promise.all(
+			settings.map(setting => presence.getSetting(setting))
+		);
+
+	switch (true) {
+		case lastActivity + IDLE_TIMEOUT < Date.now() && isIdlingPublic: {
+			presenceData.smallImageKey = SmallImageKeys.Idle;
+			presenceData.smallImageText = "Idling...";
```
Comment (theusaf) [inline websites/N/Necord/presence.ts:43]: Remove `...` from strings
```suggestion
			presenceData.smallImageText = "Idling";
```

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/N/Necord/presence.ts (hunk 5 above)
Comment (theusaf) [reply websites/N/Necord/presence.ts]: Note that this also applies to the other cases of this.

File: websites/N/Necord/metadata.json (hunk 1 above)
Comment (SocketSomeone) [reply websites/N/Necord/metadata.json]: Added privacy mode setting

File: websites/N/Necord/presence.ts (hunk 5 above)
Comment (SocketSomeone) [reply websites/N/Necord/presence.ts]: Applied at #[ee33ea4](https://github.com/PreMiD/Presences/pull/7215/commits/ee33ea48b78bb03ff93f780ddb1769b474ae7d70)

**Review by Bas950 — APPROVED** [review]

**Review by RisingSunLight42 — APPROVED** [review]

**Comment by SocketSomeone** [comment]
Fix array fill to false when `privacy: true`

**Comment by SocketSomeone** [comment]
@EncryptedDev any thinks?

**Review by Timeraa — APPROVED** [review]

**Pull request merged**